api/v1beta1/kafkacluster_types.go

```diff
@@ -544,7 +557,15 @@ type ExternalListenerConfig struct {
 	// If accessMethod is Nodeport and externalStartingPort is set to 0 then the broker IDs are not added and the Nodeport port numbers will be chosen automatically by the K8s Service controller
 	ExternalStartingPort int32 `json:"externalStartingPort"`
 	// configuring AnyCastPort allows kafka cluster access without specifying the exact broker
+	// If not defined, 29092 will be used for external clients to reach the kafka cluster
 	AnyCastPort *int32 `json:"anyCastPort,omitempty"`
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=65535
+	// +optional
+	// IngressControllerTargetPort defines the container port that the ingress controller uses for handling external traffic.
+	// If defined, IngressControllerTargetPort should be >= 1024 when using IstioIngress as the ingress controller.
+	// If not defined, 29092 will be used as the default IngressControllerTargetPort value.
+	IngressControllerTargetPort *int32 `json:"ingressControllerTargetPort,omitempty"`
```

Comment:
**Question**: Wouldn't it be easier to set the validation minimum port to 1025 (as`MaxWellKnownPort` + 1) and that would work for all ingress cases.
Or do we want to explicitly support non-IstioIngress cases to use ports from the [0;1024] range?

Reply:
Especially considering the envoy case only working because of the root setting, I would say we can get by without <=1024 ports IMO.

Reply:
Done in https://github.com/banzaicloud/koperator/pull/998/commits/7e365c5d0274157a0028cb70db2db43afedeefc1